src/test/java/co/rsk/bitcoinj/script/RedeemScriptParserFactoryTest.java

```diff
@@ -237,4 +238,45 @@ public void create_RedeemScriptParser_object_from_hardcoded_testnet_redeem_scrip
 
         Assert.assertEquals(MultiSigType.NO_MULTISIG_TYPE, parser.getMultiSigType());
     }
+
+    // TODO: Once RedeemScriptFactory get refactored and only accept redeem script type
+    //  this test should fail and it will need to be update to throw an ScriptException
+    @Test
+    public void createRedeemScriptParser_whenScriptSig() {
+        Script redeemScript = RedeemScriptUtils.createStandardRedeemScript(defaultRedeemScriptKeys);
+
+        Script p2SHOutputScript = ScriptBuilder.createP2SHOutputScript(redeemScript);
+        Script scriptSig = p2SHOutputScript.createEmptyInputScript(null, redeemScript);
+
+        Exception actualException = null;
+        try {
+            RedeemScriptParser redeemScriptParser = RedeemScriptParserFactory.get(
+                scriptSig.getChunks());
+            Assert.assertTrue(redeemScriptParser instanceof StandardRedeemScriptParser);
+        } catch (Exception ex) {
+            actualException = ex;
+        } finally {
+            Assert.assertNull(actualException);
```

Comment:
It seems as if we are not expecting this to throw an exception. In that case I would just take out the try-catch handling logic since it can be a bit confusing.

Reply:
Updated